src/Umbraco.Web/PublishedCache/NuCache/ContentStore.cs

```diff
@@ -235,11 +235,17 @@ public void ReleaseLocalDb()
             var lockInfo = new WriteLockInfo();
             try
             {
-                Lock(lockInfo);
-
-                if (_localDb == null) return;
-                _localDb.Dispose();
-                _localDb = null;
+                try{
```

Comment:
For this you mentioned

> The lock call could throw an exception in the Lock attempting to do a Monitor.Enter that is not caught and won't cause the nucache db  to be properly closed, suggest you add a catch and always dispose  of the _localDB

Why would the Lock call throw? And if it did we would have seen this exception occuring in the past but we've never seen this happen. 

Also, this code is now removing the disposal of the _localDb **unless** an exception occurs which is not what we want. It might also be possible that this whole thing is wrapped in a try/finally because disposing the _localDb might throw?

Reply:
Apologies - you are correct - it should have been in a finally - i have updated the code.
The interface signature on the Monitor.Enter shows it can throw exceptions - so we should handle them to make sure _localDB is always cleaned up. Yes disposing might throw so I've now wrapped that.